lib/rules/valid-next-tick.js

```diff
@@ -0,0 +1,167 @@
+// Helpers
+// ------------------------------------------------------------------------------
+
+/**
+ * @param {Identifier} identifier
+ * @param {RuleContext} context
+ * @returns {ASTNode|undefined}
+ */
+function getVueNextTickParentNode(identifier, context) {
+  // Instance API: this.$nextTick()
+  if (
+    identifier.name === '$nextTick' &&
+    identifier.parent.type === 'MemberExpression' &&
+    utils.isThis(identifier.parent.object, context)
+  ) {
+    return identifier.parent.parent
+  }
+
+  // Vue 2 Global API: Vue.nextTick()
+  if (
+    identifier.name === 'nextTick' &&
+    identifier.parent.type === 'MemberExpression' &&
+    identifier.parent.object.type === 'Identifier' &&
+    identifier.parent.object.name === 'Vue'
+  ) {
+    return identifier.parent.parent
+  }
+
+  // Vue 3 Global API: import { nextTick as nt } from 'vue'; nt()
+  const variable = findVariable(context.getScope(), identifier)
+
+  if (variable != null && variable.defs.length === 1) {
+    const def = variable.defs[0]
+    if (
+      def.type === 'ImportBinding' &&
+      def.node.type === 'ImportSpecifier' &&
+      def.node.imported.type === 'Identifier' &&
+      def.node.imported.name === 'nextTick' &&
+      def.node.parent.type === 'ImportDeclaration' &&
+      def.node.parent.source.value === 'vue'
+    ) {
+      return identifier.parent
+    }
+  }
+
+  return undefined
+}
+
+/**
+ * @param {CallExpression} callExpression
+ * @returns {boolean}
+ */
+function isAwaitedPromise(callExpression) {
+  return (
+    callExpression.parent.type === 'AwaitExpression' ||
+    (callExpression.parent.type === 'MemberExpression' &&
+      callExpression.parent.property.type === 'Identifier' &&
+      callExpression.parent.property.name === 'then')
+  )
+}
+
+/**
+ * @param {ASTNode} node
+ * @returns {FunctionExpression|undefined}
+ */
+function getClosestFunction(node) {
+  while (node.parent !== null) {
+    node = /** @type {ASTNode} */ (node.parent)
+
+    if (node.type === 'FunctionExpression') {
+      return node
+    }
+  }
+
+  return undefined
+}
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'enforce valid `nextTick` function calls',
+      categories: ['vue3-essential', 'essential'],
+      url: 'https://eslint.vuejs.org/rules/valid-next-tick.html'
+    },
+    fixable: 'code',
+    schema: []
+  },
+  /** @param {RuleContext} context */
+  create(context) {
+    return utils.defineVueVisitor(context, {
+      /** @param {Identifier} node */
+      Identifier(node) {
+        const parentNode = getVueNextTickParentNode(node, context)
+        if (!parentNode) {
+          return
+        }
+
+        if (parentNode.type !== 'CallExpression') {
+          context.report({
```

Comment:
I don't think we need to report this.
The user may store nextTick in a variable.

```js
const nt = Vue.nextTick
```

Reply:
I think it would be good to catch cases like `await this.$nextTick` instead of `await this.$nextTick()`, as it can be quite hard to find. Also, do we really need to consider assignment expressions? I don't think `const nt = Vue.nextTick` is a common pattern.

Reply:
It's not as common as you say. But `const nt = Vue.nextTick` is clearly looks intentional. I think it needs to be excluded.
I think we should check `await this.$nextTick` and` this.$nextTick.then()`.